### PR TITLE
Small Fragment visual update

### PR DESCRIPTION
**Pull request by TaculoTaculo**
## About The Pull Request
Adds a subtle bopping animation to fragment on containment.
Adds a animation to fragments song.
Adds particles to fragments song.

## Why It's Good For The Game
Fragment needed a bit of visual oomph and this pr tries to achieve that.

## Changelog
:cl:
tweak: Added an animation to fragments containment sprite, added a animation to fragments song.
imageadd: added icons for fragments song and his new animations
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm [hunk 1]
```diff
@@ -64,6 +69,11 @@
 	fragment.song()
 	return TRUE
 
+/mob/living/simple_animal/hostile/abnormality/fragment/Destroy()
+	QDEL_NULL(legs)
+	QDEL_NULL(particle_notes)
+	QDEL_NULL(particle_song)
+	. = ..()
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm:76]:
```suggestion
	return ..()
```

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm [hunk 2]
```diff
@@ -81,16 +91,42 @@
...
 			if(L.stat == DEAD)
 				continue
 			L.apply_damage(song_damage, WHITE_DAMAGE, null, L.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
 		SLEEP_CHECK_DEATH(3)
+	animate(src, pixel_y = 0, time = 0)
+	icon_state = "fragment_breach"
+	pixel_y = 0
+	QDEL_NULL(legs)
+	QDEL_NULL(particle_notes)
+	QDEL_NULL(particle_song)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm:129]: Don't qdel the particles, let them naturally fade. It should look better that way.

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm [hunk 3]
```diff
@@ -64,6 +69,11 @@
 	fragment.song()
 	return TRUE
 
+/mob/living/simple_animal/hostile/abnormality/fragment/Destroy()
+	QDEL_NULL(legs)
+	QDEL_NULL(particle_notes)
+	QDEL_NULL(particle_song)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm:75]: Same with the previous comment, these naturally fading would probably look better.

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm (hunk 3 above)
Comment (LanceSmites328) [reply code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm:75]: You can now rebase and call `fadeout()` on the particle_emitter's and they shall naturally fade away

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm (hunk 2 above)
Comment (TaculoTaculo) [reply code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm]: Done, added fadeout

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm (hunk 3 above)
Comment (TaculoTaculo) [reply code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm]: Same here

File: code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm (hunk 1 above)
Comment (TaculoTaculo) [reply code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm]: Oops

**Review by LanceSmites328 — APPROVED** [review]

**Pull request merged**